### PR TITLE
fixFreqForZeus

### PR DESCRIPTION
**Pull request by AdlerSalbei**
Problem was that the get SW & LR Radio command is local, this caused all units to display the Zeus frequencies.

### CODE REVIEW

**Comment by AdlerSalbei** [comment]
fixes #118

File: addons/zeus/XEH_preClientInit.sqf [hunk 1]
```diff
@@ -29,3 +29,83 @@ Place a container (like an ammo box) somewhere on the map. Fill it to your likin
...
+        };
+    }
+] call CBA_fnc_addEventHandler;
+
+[
+    "TFAR_event_OnFrequencyChanged",
+     {
+         params ["_unit", "_radio", "_channel", "", "_frequency"];
+
+         systemChat format ["Unit: %1, Channel: %2, Freq: %3", _unit, _channel, _frequency];
```
Comment (McDiod) [inline addons/zeus/XEH_preClientInit.sqf]: reminder: remove before merge

File: addons/zeus/XEH_preClientInit.sqf [hunk 2]
```diff
@@ -29,3 +29,83 @@ Place a container (like an ammo box) somewhere on the map. Fill it to your likin
...
+         systemChat format ["Radio: %1, Backpack: %2, LRadio: %3", _radio, backpack _unit, (getunitLoadout _unit) select 9 select 2];
```
Comment (McDiod) [inline addons/zeus/XEH_preClientInit.sqf]: reminder: remove before merge

File: addons/zeus/XEH_preClientInit.sqf [hunk 3]
```diff
@@ -29,3 +29,83 @@ Place a container (like an ammo box) somewhere on the map. Fill it to your likin
...
 
 [_title,_helpText] call EFUNC(ui,addHelpRecord);
+
+[
+    "TFAR_event_OnSWchannelSet",
+    {
+        params ["_unit", "_asd", "_channel", "_additional"];
+
+        private _frequency = [(call TFAR_fnc_activeSwRadio), (_channel +1)] call TFAR_fnc_getChannelFrequency;
+        systemChat format ["Unit: %1, Channel: %2, Additional: %3, radio: %4, Freq: %5", _unit, _channel, _additional, _asd, _frequency];
```
Comment (McDiod) [inline addons/zeus/XEH_preClientInit.sqf]: reminder: remove before merge

File: addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf [hunk 4]
```diff
@@ -4,27 +4,23 @@ params [["_unit",objNull]];
...
 
-private _activeSWRadio = call TFAR_fnc_ActiveSwRadio;
-if (!isNil "_activeSWRadio") then {
-    private _mainFreq = [_activeSWRadio,(_activeSWRadio call TFAR_fnc_getSwChannel) + 1] call TFAR_fnc_getChannelFrequency;
-    private _altFreq = [_activeSWRadio,(_activeSWRadio call TFAR_fnc_getAdditionalSwChannel) + 1] call TFAR_fnc_getChannelFrequency;
-
-    if (_altFreq == "") then {
-        _freqs pushBack format ["SR %1 MHz",_mainFreq];
+private _swFreq = _unit getVariable [QGVAR(freqSW), 0];
+if (!(isNil "_swFreq") && {!(_swFreq isEqualTo 0)}) then {
```
Comment (McDiod) [inline addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf]: no need to check for nil here if you are using `getVariable` with default value

File: addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf [hunk 5]
```diff
@@ -4,27 +4,23 @@ params [["_unit",objNull]];
...
+    private _swAdditionalFreq = _unit getVariable [QGVAR(freqAdditionalSW), 0];
+    if (!(isNil "_swAdditionalFreq") && {!(_swAdditionalFreq isEqualTo 0)}) then {
```
Comment (McDiod) [inline addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf]: see above

File: addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf [hunk 6]
```diff
@@ -4,27 +4,23 @@ params [["_unit",objNull]];
...
 
-private _activeLRRadio = call TFAR_fnc_activeLRRadio;
-if (!isNil "_activeLRRadio") then {
-    private _mainFreq = [_activeLRRadio,(_activeLRRadio call TFAR_fnc_getLrChannel) + 1] call TFAR_fnc_getChannelFrequency;
-    private _altFreq = [_activeLRRadio,(_activeLRRadio call TFAR_fnc_getAdditionalLrChannel) + 1] call TFAR_fnc_getChannelFrequency;
-
-    if (_altFreq == "") then {
-        _freqs pushBack format ["LR %1 MHz",_mainFreq];
+private _lrFreq = _unit getVariable [QGVAR(freqLR), 0];
+if (!(isNil "_lrFreq") && {!(_lrFreq isEqualTo 0)}) then {
```
Comment (McDiod) [inline addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf]: see above

File: addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf [hunk 7]
```diff
@@ -4,27 +4,23 @@ params [["_unit",objNull]];
...
+    private _lrAdditionalFreq = _unit getVariable [QGVAR(freqAdditionalLR), 0];
+    if (!(isNil "_lrAdditionalFreq") && {!(_lrAdditionalFreq isEqualTo 0)}) then {
```
Comment (McDiod) [inline addons/zeus/functions/fnc_moduleDiagnosticsGetPlayerFreqs.sqf]: see above

**Comment by McDiod** [comment]
does this still need testing?

**Comment by AdlerSalbei** [comment]
Not really, just somebody who approves it 🗡

**Pull request merged**